### ISSUE TITLE
Form state handling

### ISSUE THREAD

**Issue by mfal** [mittwald/flow#325]
Form states such as

- (async) validation
- (async) submission

should be indicated to the user.

**Comment by mfal** [comment]
Related MR https://github.com/mittwald/flow/pull/493
